### PR TITLE
Metrics for classification tasks

### PR DESCRIPTION
**Pull request by Oziel14**
# Summary
This branch depends on the PR #77 .
The metrics class was created as well as a register for them.
Incorporated accuracy, recall, precision and f1score metrics.
Tests were made for the metrics

## Type of change
- Back end new feature.

## Changes
- Changes in core/config files

## How to Test
Tests were performed on the test_classification_metrics.py

## Notes
By approving this PR you would already have a complete back end pipeline to train and evaluate a model. But only the task of tabulating classification, sklearn models and the mentioned metrics.

### CODE REVIEW

File: DashAI/back/metrics/Classification/f1score.py
Comment (pbadillatorrealba) [inline DashAI/back/metrics/Classification/f1score.py]: Both branches yields the same results! (check the average parameter).
The same applies for the rest of the metrics.

Note that there are also several ways to average by having multi-classes.
See more in the [f1 documentation](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.f1_score.html).

File: DashAI/back/metrics/Classification/accuracy.py
Comment (pbadillatorrealba) [inline DashAI/back/metrics/Classification/accuracy.py]: This function is often repeated. Why not put it in `ClassificationMetric` and then have the subclass simply inherit it?

File: DashAI/back/metrics/Classification/f1score.py
Comment (pbadillatorrealba) [inline DashAI/back/metrics/Classification/f1score.py]: `F1` or `F1Score` ?

The following metrics does not have that suffix.
I think it is good to standardize this at this point.

File: DashAI/back/metrics/Classification/precision.py
Comment (pbadillatorrealba) [inline DashAI/back/metrics/Classification/precision.py]: Same error as before.

File: DashAI/back/metrics/Classification/recall.py
Comment (pbadillatorrealba) [inline DashAI/back/metrics/Classification/recall.py]: Same error as before.

**Review by pbadillatorrealba — CHANGES_REQUESTED** [review]
I am in doubt.

Should we treat the metric as a configurable object?
Or a static one, like the tasks?

I ask because for the case of precision, recall and f1 in multiclass configuration have a parameter to configure how the average of the different values of the metric per class (micro, macro, weighted, etc...) will be calculated.

I particularly like the first option, but for simplicity, I would go for the second one and always keep the calculation fixed, making different F1 classes such as F1_micro, F1_macro, F1_weighted, etc...

What do you think of the options?

Another issue: seeing that the registry is close to being approved, merge the current metrics registry could incur in bugs. 
Would you be okay if you wait for the new log to be merged before merging this branch?

**Comment by Oziel14** [comment]
About the first remark the idea is that it is a configurable object, in fact the base class of metrics comes from the class of configurable objects. But due to the time for now it will be static. And that explains that metrics like precision or recall for multiclasses use micro average.

And about the second point, there is no problem with waiting for the change of the registry before merging this branch

**Review by pbadillatorrealba — APPROVED** [review]
Looks great!

**Review by cristian-tamblay — CHANGES_REQUESTED** [review]
We should change `DashAI/back/metrics/Classification` folder name to `DashAI/back/metrics/classification` so we don't have to fix it in the future.

**Comment by Oziel14** [comment]
Name change done

**Review by cristian-tamblay — APPROVED** [review]

**Pull request merged**
